src/runtime_src/core/tools/xbmgmt2/SubCmdAdvanced.cpp

```diff
@@ -0,0 +1,143 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "SubCmdAdvanced.h"
+#include "OO_Clock.h"
+#include "OO_Config.h"
+#include "OO_NIFD.h"
+
+#include "common/system.h"
+#include "common/device.h"
+#include "common/xclbin_parser.h"
+
+#include "tools/common/XBUtilities.h"
+namespace XBU = XBUtilities;
+
+// 3rd Party Library - Include Files
+#include <boost/program_options.hpp>
+namespace po = boost::program_options;
+
+// System - Include Files
+#include <iostream> 
+
+#include "common/system.h"
+#include "common/device.h"
+#include <boost/format.hpp>
+#include <map>
+
+// ----- C L A S S   M E T H O D S -------------------------------------------
+
+SubCmdAdvanced::SubCmdAdvanced(bool _isHidden, bool _isDepricated, bool _isPreliminary)
+    : SubCmd("advanced", 
+             "Low level command operations")
+{
+  const std::string longDescription = "Low level command operations.";
+  setLongDescription(longDescription);
+  setExampleSyntax("");
+  setIsHidden(_isHidden);
+  setIsDeprecated(_isDepricated);
+  setIsPreliminary(_isPreliminary);
+}
+
+
+void
+SubCmdAdvanced::execute(const SubCmdOptions& _options) const
+{
+  XBU::verbose("SubCommand: advanced");
+
+  // ============= Define all of the options =================================
+  po::options_description allDesc;
+
+  // -- Common top level options ---
+  bool help = false;
+
+  po::options_description commonTopDesc("Common Options"); 
+  commonTopDesc.add_options()
+    ("help,h", boost::program_options::bool_switch(&help), "Help to use this sub-command")
+  ;
+  allDesc.add(commonTopDesc);
+
+  // -- Define the supporting option options ----
+  SubOptionOptions subOptionOptions;
+  subOptionOptions.emplace_back(std::make_shared<OO_Clock>("clock"));
+  subOptionOptions.emplace_back(std::make_shared<OO_Config>("config"));
+  subOptionOptions.emplace_back(std::make_shared<OO_NIFD>("nifd"));
+
+  for (auto subOO : subOptionOptions) {
+    allDesc.add_options()(subOO->longName().c_str(), subOO->description().c_str());
+    subOO->setExecutable(getExecutableName());
+    subOO->setCommand(getName());
+  }
+
+  // =========== Process the options ========================================
+
+  // 1) Process the common top level options 
+  po::parsed_options parsedCommonTop = 
+    po::command_line_parser(_options).
+    options(allDesc).          
+    allow_unregistered().           // Allow for unregistered options
+    run();                          // Parse the options
+
+  po::variables_map vm;
+
+  try {
+    po::store(parsedCommonTop, vm);  // Can throw
+    po::notify(vm);                  // Can throw (but really isn't used)
+
+    // Multual DRC
+    for (unsigned index1 = 0; index1 < subOptionOptions.size(); ++index1) {
```

Comment:
unsigned apples ? :-)
When I see for loops over indices where indices are used to index some struct, then I think there must be a std::algorithm for this.

Reply:
Done